cmd/topology/topology_test.go

```diff
@@ -0,0 +1,248 @@
+package topology
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	tfake "github.com/google/kne/api/clientset/v1beta1/fake"
+	topopb "github.com/google/kne/proto/topo"
+	"github.com/google/kne/topo"
+	"github.com/google/kne/topo/node"
+	"github.com/h-fam/errdiff"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/encoding/prototext"
+	kfake "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+)
+
+func NewNR(pb *topopb.Node) (node.Implementation, error) {
+	return &notResetable{pb: pb}, nil
+}
+
+type notResetable struct {
+	pb            *topopb.Node
+	configPushErr error
+}
+
+func (nr *notResetable) Proto() *topopb.Node {
+	return nr.pb
+}
+
+func (nr *notResetable) CreateNodeResource(_ context.Context, _ node.Interface) error {
+	return status.Errorf(codes.Unimplemented, "Unimplemented")
+}
+
+func (nr *notResetable) DeleteNodeResource(_ context.Context, _ node.Interface) error {
+	return status.Errorf(codes.Unimplemented, "Unimplemented")
+}
+
+func (nr *notResetable) ConfigPush(_ context.Context, s string, r io.Reader) error {
+	b, err := ioutil.ReadAll(r)
+	if err != nil {
+		return err
+	}
+	if string(b) == "error" {
+		return fmt.Errorf("error")
+	}
+	return nr.configPushErr
+}
+
+type resetable struct {
+	*notResetable
+	resetErr error
+}
+
+func (r *resetable) ResetCfg(ctx context.Context, ni node.Interface) error {
+	return r.resetErr
+}
+
+func NewR(pb *topopb.Node) (node.Implementation, error) {
+	return &resetable{&notResetable{pb: pb}, nil}, nil
+}
+
+func writeTopology(t *testing.T, topo *topopb.Topology) (*os.File, func()) {
+	t.Helper()
+	f, err := ioutil.TempFile("", "reset")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer os.Remove(t.Name())
```

Comment:
Is this line needed? Not sure what exactly is being removed, the test name? The file cleanup func is returned so not sure

Reply:
no idea where that came frome...